### DISCUSSION TITLE
Maximum send chat

### DISCUSSION THREAD

**Discussion by kingter-sutjiadi** [inspircd/inspircd#2077] [Support]
Hi firstly i want to thank for this awesome open source project, anyway i want to ask can i modify the maximum send chat size on the server, i'm planning to use json format when sending message, i already forked this project and test some changes but when my custom client tried to send chat more than 433 the other client can't see the rest of the message it was trimmed,
Thank you

**Comment by sadiepowell** [comment]
You can increase `<limits:maxline>` (default: 512) to increase the line length if you're only using custom clients. Please note that this *will* break normal IRC clients though.

You might also find it easier to use the IRCv3 client-to-client tags and `TAGMSG` to send data between clients as the client-to-client tag space is 4x larger than what you can use for regular messages. For this you need the [`cap`](https://docs.inspircd.org/3/modules/cap/) and [`ircv3_ctctags`](https://docs.inspircd.org/3/modules/ircv3_ctctags/) modules. See the [IRCv3 message tags specification](
https://ircv3.net/specs/extensions/message-tags.html) for more information on how to use this.

**Reply by kingter-sutjiadi** [reply]
Hi thank you @SadieCat appreciate your fast responses,

btw yeah i'm using for custom clients 

Tried and success Cheers

**Reply by kingter-sutjiadi** [reply]
if you mind i wanted to ask something else about hardware requirements, i'm planning to have concurrent approximately around 10k - 20k 
i'm planning on doing horizontal scaling and around 3/ 4 instances what's the recommendation hardware spec for running the inspircd?

and normally how many concurrent connection for single instances to handle?

Thank you

**Reply by sadiepowell** [reply]
I've seen InspIRCd networks easily scale up to 100k users on a single server but you should probably have multiple servers running for redundancy in case one goes down.

**Reply by kingter-sutjiadi** [reply]
ok cool thank you, may i know what's the server spesification ? will it enough with 2 CPU and 4 GB ram? or it need more that this?

**Reply by sadiepowell** [reply]
I don't remember but that's probably enough.

**Reply by kingter-sutjiadi** [reply]
thanks a lot for your help appreciate @SadieCat 👍 🙏
